File: two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ans = []
        for i in nums:
            if target-i in nums:
                if nums.index(i) != nums.index(target-i):
                    ans.append(nums.index(i))
                    ans.append(nums.index(target-i))
                    break
                else:
                    if nums.count(i) > 1 :
                        ans.append(nums.index(i))
                        nums.remove(i)
                        ans.append(nums.index(i)+1)
                        break
        return ans

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashMap;
        
        for (int i = 0; i < nums.size(); ++i) {
            int diff = target - nums[i];
            
            // Check if the complement exists in the map
            if (hashMap.find(diff) != hashMap.end()) {
                return {hashMap[diff], i};
            }
            
            // Store the index of the current number
            hashMap[nums[i]] = i;
        }
        
        return {}; // Return an empty vector if no solution is found
    }
};